spec/requests/questions_spec.rb

```diff
@@ -1,148 +1,233 @@
 require "rails_helper"
 
 RSpec.describe "Questions", type: :request do
-  let(:quiz) { Quiz.create!(title: "Sample Quiz", description: "This is a sample quiz") }
+  before do
+    login_as(user, scope: :user)
+  end
+
+  let(:user) { User.create!(email: "test@example.com", password: "password") }
+  let(:quiz) { Quiz.create!(title: "Sample Quiz", description: "This is a sample quiz", user: user) }
 
   describe "GET new" do
-    it "renders the new question form" do
-      get new_quiz_question_path(quiz)
-      expect(response).to have_http_status(:success)
-      expect(response.body).to include("Add a New Question")
+    context "when the user owns the quiz" do
+      it "renders the new question form" do
+        get new_quiz_question_path(quiz)
+        expect(response).to have_http_status(:success)
+        expect(response.body).to include("Add a New Question")
+      end
+    end
+
+    context "when the user does not own the quiz" do
+      let(:other_user) { User.create!(email: "other@example.com", password: "password") }
+      let(:other_quiz) { Quiz.create!(title: "Other Quiz", description: "Another quiz", user: other_user) }
+
+      it "returns a 404 response" do
+        get new_quiz_question_path(other_quiz)
+        expect(response).to have_http_status(404)
```

Comment:
```suggestion
        expect(response).to have_http_status(:not_found)
```

In your successful tests we assert on getting a success response code rather than the actual response, so it'd be consistent for the client error status codes to also use the symbol means to assert.

Note I've only left one comment but there's a few to catch (find and replace should do it)

Reply:
That makes sense, yeah! I've replaced them all onw